src/cephadm/cephadm

```diff
@@ -1515,6 +1515,7 @@ class CephContainer:
             'run',
             '--rm',
             '--net=host',
+            '--cap-add=SYS_PTRACE',
```

Comment:
Well, that's a big security hole. like super huge. I think we have to make this opt-in :cry:

Reply:
For core dumps, adding `--ulimit core=-1` to podman commands should work.

Reply:
Can you clarify what the security implications of this are?  My understanding is that it means you can strace/gdb attach to processes inside the container, but not processes outside the container.  Is the risk/concern here that if someone breaks out of the container so that they can access a process in a different namespace, then they could observe that process?

From my perspective that is fine.  IMO cephadm is using containers primarily as a tool to ease provisioning, not as a means to improve security (although there are obviously many security benefits we get along the way).

Reply:
Hm. We'd fall behind the current security capabilities of standard distributions. As a normal user, you no longer always have  `SYS_PTRACE`. By using containers, users will expect to have a tighter security context compared to standard unix users. 

By unconditionally adding SYS_PTRACE to all production clusters, we'd surprise users, which seems to be problematic to me. 

What about making this dependent on a environment variable?